### PR TITLE
Che 7 typescript tests

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
This PR adds typescript tests of Che 7. These tests are based on upstream tests and should replace current Che 7 tests. This PR is WIP - I'll refactor that, add headers to files and remove comments, but I would like to hear your opinions on that meantime. 

### How have you tested this PR?
localy against production

### CODE REVIEW

File: theia-tests/inversify.config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { ILoginPage } from 'e2e';
+import { TYPES, CLASSES } from 'e2e';
+import { RhCheLoginPage } from './pageobjects/RhCheLoginPage';
+import { e2eContainer } from 'e2e';
```
Comment (jrichter1) [inline theia-tests/inversify.config.ts]: you can group up imports from the same package into one line
```typescript
import { x, y, z } from 'package'
```

File: theia-tests/tslint.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        140
+      ],
+      "member-ordering": [
+        true,
+        "public-before-private",
+        "static-before-instance",
+        "variables-before-functions"
+      ],
+      "no-arg": true,
+      "no-bitwise": true,
```
Comment (ppitonak) [inline theia-tests/tslint.json]: I suggest adding `"no-consecutive-blank-lines": true,` ;)

File: theia-tests/pageobjects/RhCheLoginPage.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    constructor(
+        @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper,
+        @inject(TYPES.Driver) private readonly driver: IDriver) { }
+
+    async login(timeout: number = 20) {
+        const webDriver: ThenableWebDriver = this.driver.get();
+        await webDriver.navigate().to(TestConstants.TS_SELENIUM_BASE_URL);
+
+        const usernameFieldLocator: By = By.id(USERNAME_INPUT_ID);
+        await this.driverHelper.waitVisibility(By.id(USERNAME_INPUT_ID), 20000);
```
Comment (jrichter1) [inline theia-tests/pageobjects/RhCheLoginPage.ts]: use the ```usernameFieldLocator``` constant you just declared

File: theia-tests/pageobjects/RhCheLoginPage.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        await this.driverHelper.enterValue(usernameFieldLocator, TestConstants.TS_SELENIUM_USERNAME, timeout );
+
+        const nextButtonLocator: By = By.id(NEXT_BUTTON_ID);
+        await this.driverHelper.waitAndClick(nextButtonLocator, timeout);
+
+        const passwordFieldLocator: By = By.id(PASSWORD_INPUT_ID);
+        await this.driverHelper.waitVisibility(By.id(PASSWORD_INPUT_ID), 3000);
```
Comment (jrichter1) [inline theia-tests/pageobjects/RhCheLoginPage.ts]: use ```passwordFieldLocator``` constant

File: theia-tests/package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "lint": "tslint --fix -p .",
+    "set_env": "export TS_SELENIUM_LOAD_PAGE_TIMEOUT=300000",
+    "debug": "npm run tsc && mocha --opts mocha.opts"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/redhat-developer/rh-che.git"
+  },
+  "author": "Katerina Foniok (kkanova@redhat.com)",
+  "license": "ISC",
```
Comment (ppitonak) [inline theia-tests/package.json]: headers in source files say EPL-2.0

File: theia-tests/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+set TS_SELENIUM_LOAD_PAGE_TIMEOUT to 300000 ms
```
Comment (ppitonak) [inline theia-tests/README.md]: please provide some context to this command :)

File: theia-tests/tests/HappyPath.spec.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+suite('RhChe E2E', async () => {
+    suite('Login and wait dashboard', async () => {
+        test('Login', async () => {
+            await loginPage.login();
+            await dashboard.waitPage(30000);
+        });
+    });
+
+    suite('Create workspace and add plugin', async () => {
+        test('Open \'New Workspace\' page', async () => {
```
Comment (jrichter1) [inline theia-tests/tests/HappyPath.spec.ts]: if you use a template string, you don't have to escape single quotes :)

File: theia-tests/package.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "theia-tests",
+  "version": "1.0.0",
+  "description": "Tests for RH-Che with Theia IDE",
+  "main": "index.ts",
+  "scripts": {
+    "test": "npm run set_env && npm run lint && npm run tsc && mocha --opts mocha.opts",
+    "tsc": "tsc -p .",
```
Comment (jrichter1) [inline theia-tests/package.json]: I'd suggest cleaning the dist folder before compiling, for added safety

File: theia-tests/tslint.json [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        true,
+        "check-branch",
+        "check-decl",
+        "check-operator",
+        "check-separator",
+        "check-type"
+      ]
+    }
+  }
+
```
Comment (ibuziuk) [inline theia-tests/tslint.json]: no new line in the end

File: theia-tests/mocha.opts [hunk 10]
```diff
@@ -0,0 +1,3 @@
+--timeout 1200000
+-u tdd
+--spec ./dist/tests/HappyPath.spec.js
```
Comment (ibuziuk) [inline theia-tests/mocha.opts]: no new line in the end

File: theia-tests/README.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+## Running the tests
+There are two ways how to run them - using docker file with pre-installed dependencies or using a script. 
+*... in progress ...*
+
+## Test flow
+The tests are end-to-end tests that should represent Happy path through a product. Current flow is following:
+
+- Login
+- Create Che 7 workspace with console-java-simple stack
```
Comment (ibuziuk) [inline theia-tests/README.md]: there is no `console-java-simple`, please use `java-maven` stack for testing

[1] https://github.com/redhat-developer/rh-che/blob/master/assembly/fabric8-stacks/src/main/resources/stacks.json#L330

Comment (Katka92) [reply theia-tests/README.md]: That should be Che 7 stack and console-java-simple project. Thanks for noticing

**Pull request merged**